skimage/measure/tests/test_marching_cubes.py

```diff
@@ -74,5 +96,68 @@ def test_correct_mesh_orientation():
 
 
+def test_both_algs_same_result_ellipse():
+    # Performing this test on data that does not have ambiguities
+    
+    sphere_small = ellipsoid(1, 1, 1, levelset=True)
+    
+    vertices1, faces1 = marching_cubes(sphere_small, 0)[:2]
+    vertices2, faces2 = marching_cubes_lewiner(sphere_small, 0, allow_degenerate=False)[:2]
+    vertices3, faces3 = marching_cubes_lewiner(sphere_small, 0, allow_degenerate=False, use_classic=True)[:2]
+    
+    # Order is different, best we can do is test equal shape and same vertices present
+    assert _same_mesh(vertices1, faces1, vertices2, faces2)
+    assert _same_mesh(vertices1, faces1, vertices3, faces3)
+
+
+def _same_mesh(vertices1, faces1, vertices2, faces2, tol=1e-10):
+    """ Compare two meshes, using a certain tolerance and invariant to
+    the order of the faces.
+    """
+    # Unwind vertices
+    triangles1 = vertices1[np.array(faces1)]
+    triangles2 = vertices2[np.array(faces2)]
+    # Sort vertices within each triangle
+    triang1 = [np.concatenate(sorted(t, key=lambda x:tuple(x))) for t in triangles1]
+    triang2 = [np.concatenate(sorted(t, key=lambda x:tuple(x))) for t in triangles2]
+    # Sort the resulting 9-element "tuples"
+    triang1 = np.array(sorted([tuple(x) for x in triang1]))
+    triang2 = np.array(sorted([tuple(x) for x in triang2]))
+    return triang1.shape == triang2.shape and np.allclose(triang1, triang2, 0, tol)
+
+
+def test_both_algs_same_result_donut():
+    # Performing this test on data that does not have ambiguities
+    
+    n = 48
+    a, b = 2.5/n, -1.25
+    isovalue = 0.0
+    #
+    vol = np.empty((n,n,n), 'float32')
+    for iz in range(vol.shape[0]):
+        for iy in range(vol.shape[1]):
+            for ix in range(vol.shape[2]):
+                # Double-torii formula by Thomas Lewiner
+                z, y, x = float(iz)*a+b, float(iy)*a+b, float(ix)*a+b
+                vol[iz,iy,ix] = ( ( 
+                    (8*x)**2 + (8*y-2)**2 + (8*z)**2 + 16 - 1.85*1.85 ) * ( (8*x)**2 +
+                    (8*y-2)**2 + (8*z)**2 + 16 - 1.85*1.85 ) - 64 * ( (8*x)**2 + (8*y-2)**2 )
+                    ) * ( ( (8*x)**2 + ((8*y-2)+4)*((8*y-2)+4) + (8*z)**2 + 16 - 1.85*1.85 )
+                    * ( (8*x)**2 + ((8*y-2)+4)*((8*y-2)+4) + (8*z)**2 + 16 - 1.85*1.85 ) -
+                    64 * ( ((8*y-2)+4)*((8*y-2)+4) + (8*z)**2 
+                    ) ) + 1025
+    
+    vertices1, faces1 = marching_cubes(vol, 0)[:2]
+    vertices2, faces2 = marching_cubes_lewiner(vol, 0)[:2]
+    vertices3, faces3 = marching_cubes_lewiner(vol, 0, use_classic=True)[:2]
+    
+    # Old and new alg are different
+    assert not _same_mesh(vertices1, faces1, vertices2, faces2)
+    # New classic and new Lewiner are different
+    assert not _same_mesh(vertices2, faces2, vertices3, faces3)
+    # Would have been nice if old and new classic would have been the same
+    # assert _same_mesh(vertices1, faces1, vertices3, faces3, 5)
```

Comment:
I found a single typo above in the Lewiner implementation's classic pathway - after fixing that, revisit this and see if they agree. That would certainly make life simpler...